### PR TITLE
Add summary section for local kubernetes

### PR DESCRIPTION
**Pull request by iocanel**
For some people it seems that the options for local kubernetes deployment are not obvious. So, let's add a section that summarizes the options for local kubernetes deployment and leads in to the sections for minikube and kind.

### CODE REVIEW

**Review by geoand — APPROVED** [review]

File: docs/src/main/asciidoc/deploying-to-kubernetes.adoc [hunk 1]
```diff
@@ -925,7 +925,25 @@ quarkus.kubernetes.rbac.cluster-role-bindings.my-cluster-role-binding.role-name=
...
+- Setting `imagePullPolicy` to `IfNotPresent`
+- Using `NodePort` as `Service` type
+
+Quakrus proivdes extensions that among others set these options by default.
+Such extensions are:
+
+- quarkus-minikube
+- quarkus-kind
+
+If the list of extensions does not match the tool you are using (e.g. docker desktop, microk8s etc) then its suggested to use the `quarkus-minikube` extension
```
Comment (gsmet) [inline docs/src/main/asciidoc/deploying-to-kubernetes.adoc:942]:
```suggestion
If the list of extensions does not match the tool you are using (e.g. Docker Desktop, microk8s etc) then it is suggested to use the `quarkus-minikube` extension.
```

File: docs/src/main/asciidoc/deploying-to-kubernetes.adoc [hunk 2]
```diff
@@ -925,7 +925,25 @@ quarkus.kubernetes.rbac.cluster-role-bindings.my-cluster-role-binding.role-name=
...
+as it's default should be reasonable for most environments.
```
Comment (gsmet) [inline docs/src/main/asciidoc/deploying-to-kubernetes.adoc:943]:
```suggestion
as its defaults should be reasonable for most environments.
```

File: docs/src/main/asciidoc/deploying-to-kubernetes.adoc [hunk 3]
```diff
@@ -925,7 +925,25 @@ quarkus.kubernetes.rbac.cluster-role-bindings.my-cluster-role-binding.role-name=
 <1> In this example, the cluster role "my-cluster-role" will be generated with the specified policy rules.
 <2> The name of the ClusterRole resource to use. Role resources are namespace-based and hence not allowed in ClusterRoleBinding resources.
 
-=== Deploying to Minikube
+=== Deploying to local Kubernetes
+
+When deploying to local Kubernetes environments users often perform minor changes to their manifests that simplify the development process.
```
Comment (gsmet) [inline docs/src/main/asciidoc/deploying-to-kubernetes.adoc:930]:
```suggestion
When deploying to local Kubernetes environments, users often perform minor changes to their manifests that simplify the development process.
```

**Review by gsmet — CHANGES_REQUESTED** [review]
Spotted a few things that need fixing.

**Comment by iocanel** [comment]
@gsmet: applied your commits and squashed them. Thanks!

**Review by gsmet — APPROVED** [review]
Very nice, thanks!

**Comment by gsmet** [comment]
I squashed the commits before merging.

**Pull request merged**